packages/msgraph/src/Adaptor.js

```diff
@@ -76,14 +91,205 @@ export function create(resource, data, callback) {
  */
 export function get(path, query, callback = false) {
   return state => {
-    const resolvePath = expandReferences(path)(state);
-    const resolveQuery = expandReferences(query)(state);
     const { accessToken, apiVersion } = state.configuration;
+    const [resolvedPath, resolvedQuery] = expandReferences(state, path, query);
+
+    const url = getUrl(resolvedPath, apiVersion);
+    const auth = getAuth(accessToken);
+
+    return request(url, { ...resolvedQuery, ...auth }).then(response =>
+      handleResponse(response, state, callback)
+    );
+  };
+}
+
+/**
+ * Get a Drive or SharePoint document library. The drive metadata will be written
+ * to state.drives, where it can be used by other adaptor functions.
+ * Pass { id } to get a drive by id or { id, owner } to get default drive for
+ * some parent resource, like a group
+ * @public
+ * @example <caption>Get a drive by ID</caption>
+ * getDrive({ id: "YXzpkoLwR06bxC8tNdg71m" })
+ * @example <caption>Get the default drive for a site</caption>
+ * getDrive({ id: "openfn.sharepoint.com", owner: "sites" })
+ * @param specifier {Object} - A definition of the drive to retrieve
+ *    - id {string} - The ID of the resource or owner.
+ *    - owner {string} - The type of drive owner (e.g. sites, groups).
+ * @param {string} name - The local name of the drive used to write to state.drives, ie, state.drives[name]
+ * @param {function} [callback = s => s] (Optional) Callback function
+ * @return {Operation}
+ */
+export function getDrive(specifier, name = 'default', callback = s => s) {
+  return state => {
+    const { accessToken, apiVersion } = state.configuration;
+    const [resolvedSpecifier, resolvedName] = expandReferences(
+      state,
+      specifier,
+      name
+    );
+
+    const { id, owner = 'drive' } = resolvedSpecifier;
+
+    let resource;
+    if (owner === 'drive') {
+      resource = `drives/${id}`;
+    } else {
+      resource = `${owner}/${id}/drive`;
+    }
+
+    const url = getUrl(resource, apiVersion);
+    const auth = getAuth(accessToken);
+
+    return request(url, { ...auth }).then(response => {
+      state.drives[resolvedName] = response;
+      return callback(state);
+    });
+  };
+}
+
+/**
+ * Get folder metadata or an array of folder contents.
```

Comment:
This should say "get the contents or metadata of a folder", because contents have precedence (they are they default)

Reply:
Done [c6dee2f](https://github.com/OpenFn/adaptors/pull/297/commits/c6dee2f238fd73974bea9557342c399c735e70fd)